src/server/plugins/engine/components/ComponentCollection.ts

```diff
@@ -135,4 +196,39 @@ export class ComponentCollection {
 
     return result
   }
+
+  /**
+   * Return error by code for collection fields
+   * For example, 'date.min', 'date.max'
+   */
+  error(helpers: CustomHelpers, code: string, context?: Context) {
+    const { formItems, parent } = this
+    const [field] = formItems
+
+    // Use collection parent name/title when available
+    const key = parent?.name ?? field.name
+    const title = parent?.title ?? field.title
+
+    // Support collection parent {{#title}} in messages
+    const local: Context = {
+      title: title.toLowerCase(),
+      ...context
+    }
+
+    // Error summary links to first collection field
+    const localState: State = parent
+      ? { key, path: [parent.name, field.name] } // For example, `#dateField__day`
+      : { key, path: [field.name] } // Otherwise `#dateField` for single fields
```

Comment:
I've also fixed this issue where `href="#dateField"` in the error summary wouldn't link anywhere

Using the parent field, we now locate the first `href="#dateField__day"` when the entire date is invalid

E.g. When validating impossible dates such as 31/2/2024